### PR TITLE
Add codespell support (config, workflow to alert on new typos) + make it fix typos

### PR DESCRIPTION
**Pull request by yarikoptic**

### CODE REVIEW

**Comment by aschnell** [comment]
Thanks, could you add a link to the documentation in codespell.yml?

File: .codespellrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[codespell]
```
Comment (yarikoptic) [inline .codespellrc:1]:
```suggestion
# Ref: https://github.com/codespell-project/codespell#using-a-config-file
[codespell]
```

File: .github/workflows/codespell.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (yarikoptic) [inline .github/workflows/codespell.yml:1]:
```suggestion
# Configure via .codespellrc in the root of the project
---
```

**Comment by yarikoptic** [comment]
Added references in the config and in the workflow

**Pull request merged**